todoapp/src/components/TodoApp.vue

```diff
@@ -0,0 +1,243 @@
+import { onMounted, ref, computed } from 'vue'
+import moment from 'moment'
+import type { TodoType } from '../types/text'
+import Header from './Header.vue'
+import Todos from './Todos/Todos.vue'
+import FinishedTodos from './FinishedTodos/FinishedTodos.vue'
+import SearchTodos from './SearchTodos.vue'
+import FilterTodos from './FilterTodos.vue'
+import DeletePopup from './DeletePopup.vue'
+
+const todos = ref<TodoType[]>([])
+const finishedTodos = ref<TodoType[]>([])
+const isPopUp = ref(false)
+const isFinishedTodoPopUp = ref(false)
+const selectedTodoId = ref()
+const selectedFinishedTodoId = ref()
+const searchInput = ref('')
+const localSearchInput = ref('')
+const currentEditedTodo = ref<number | null>(null)
+
+const selectedSortingButtons = ref({
+  title: { selected: true, order: true },
+  description: { selected: false, order: true },
+  priority: { selected: false, order: true },
+  date: { selected: false, order: true }
+})
+
+const commonTodoProperties = {
+  text: 'Add a description',
+  priority: 0,
+  isPriorityChange: false,
+  isChecked: false,
+  isEditing: false
+}
+
+function addTodo() {
+  const currentDate = moment().format('DD.MM.YYYY')
+  const customId = createId()
+
+  const newTodo = {
+    title: 'Add a title',
+    createdAt: currentDate,
+    id: customId,
+    ...commonTodoProperties
+  }
+
+  todos.value.push(newTodo)
+  localStorage.setItem('todos', JSON.stringify(todos.value))
+}
+
+function addSearchTodo() {
+  const currentDate = moment().format('DD.MM.YYYY')
+  const customId = createId()
+  const searchInputValue = searchInput.value.trim()
+
+  const newTodo = {
+    title: searchInputValue,
+    createdAt: currentDate,
+    id: customId,
+    ...commonTodoProperties
+  }
+
+  todos.value.push(newTodo)
+  localStorage.setItem('todos', JSON.stringify(todos.value))
+  clearSearchInput()
+}
+
+function clearSearchInput() {
+  searchInput.value = ''
+  localSearchInput.value = ''
+}
+
+function closePopUp() {
+  isPopUp.value = false
+  isFinishedTodoPopUp.value = false
+}
+
+function createId() {
+  const timestamp = Date.now()
+  const randomNum = Math.floor(Math.random() * 1000)
+  return timestamp + randomNum
+}
+
+function displayPopup(isFinishedTodo: boolean) {
+  if (isFinishedTodo) {
+    isFinishedTodoPopUp.value = true
+  } else {
+    isPopUp.value = true
+  }
+}
+
+function doSearch() {
+  searchInput.value = localSearchInput.value
+}
+
+function getTodoId(id: number, isFinishedTodo: boolean) {
+  currentEditedTodo.value = id
+  if (isFinishedTodo) {
+    selectedFinishedTodoId.value = id
+  } else {
+    selectedTodoId.value = id
+  }
+}
+
+const isPopupDisplayed = computed(() => isPopUp.value || isFinishedTodoPopUp.value)
+
+function markTodoStatus(todo: TodoType) {
+  if (todo.isChecked) {
+    const updatedFinishedTodos = [...finishedTodos.value, todo]
+    const updatedTodos = todos.value.filter((item) => item.id !== todo.id)
+    finishedTodos.value = updatedFinishedTodos
```

Comment:
Please move this part after the if ..else, this block added to if and else as well, so you don't need to duplicate it.
`finishedTodos.value = updatedFinishedTodos; todos.value = updatedTodos`